nomad/container.py

```diff
@@ -165,18 +167,28 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_name}-{name}'.format(
```

Comment:
OK. I've added a commit to enforce container names. (3) and (4) are no longer required since the name of an implicit container (that is, a container which is not defined under the "containers" section) will automatically default to "default". I think this is the correct behavior for now.

Regarding the validation rules problem and how to "properly" enforce project names / container names (without presenting a KeyError exception to the user)... these aspects should be improved in a subsequent PR! :wink: